CoseSign1.Certificates/Local/Validators/X509ChainTrustValidator.cs

```diff
@@ -115,6 +112,23 @@ public class X509ChainTrustValidator : X509Certificate2MessageValidator
             return new CoseSign1ValidationResult(GetType(), true, "Certificate was Trusted.");
         }
 
+        // If we fail because chain build failed to reach the revocation server, retry in case the server is down.
+        if (ChainBuilder.ChainPolicy.RevocationMode != X509RevocationMode.NoCheck)
+        {
+            int maxAttempts = 3;
+            for (int i = 0;
+                i < maxAttempts && ChainBuilder.ChainStatus.Any(s => (s.Status & X509ChainStatusFlags.RevocationStatusUnknown) != 0);

```

Comment:
Does this make a distinction between failing to check CRL because of the extension not existing? If so, we should consider skipping this loop entirely and returning successfully if the [CRL X509Extension](https://learn.microsoft.com/en-us/windows/win32/seccertenroll/supported-extensions#crldistributionpoints) is absent and there are no other errors

Reply:
I don't think it makes that distinction. The only flags I can find that pertain to CRL are OfflineRevocation, RevocationStatusUnknown, and Revoked.
https://learn.microsoft.com/en-us/dotnet/api/system.security.cryptography.x509certificates.x509chainstatusflags?view=net-8.0
If there's a use case for the CRL extension being absent let's put that in a separate task and figure out what the best approach is.

Reply:
I can confirm that I've come across several intermediates that do not include this extension, although I do not know how common they are. Addressing this in a future PR sounds good to me 👍